src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -419,6 +425,10 @@ private void updateRetentionTime(final EventType original, final EventType event
             } else {
                 eventType.setOptions(original.getOptions());
             }
+            if (isRepartitionRequired.compareAndSet(true, false)) {
```

Comment:
the request is executed in one thread, why do you need to keep the state between the calls? 
I would rather reorganize the code